Pinta/MainWindow.cs

```diff
@@ -393,17 +393,56 @@ private void MainWindow_DeleteEvent (object o, DeleteEventArgs args)
 
 		private void MainWindow_DragDataReceived (object o, DragDataReceivedArgs args)
 		{
+			// TODO: Generate random name for the picture being downloaded
+
 			// Only handle URIs
 			if (args.Info != 100)
 				return;
 
 			string fullData = System.Text.Encoding.UTF8.GetString (args.SelectionData.Data);
 
-			foreach (string individualFile in fullData.Split ('\n')) {
-				string file = individualFile.Trim ();
-
-				if (file.StartsWith ("file://"))
-					PintaCore.Workspace.OpenFile (new Uri (file).LocalPath);
+			try {
+				PintaCore.Workspace.HasOpenPendingDocuments = true;
```

Comment:
IMO it should be removed - the only place it's checked is inside `WindowPointToCanvas`, but mouse events shouldn't be handled while an image is being downloaded. The progress dialog should be a modal dialog, while the image is downloaded in the background.